pythonforandroid/recipes/libxml2/__init__.py

```diff
@@ -0,0 +1,31 @@
+from pythonforandroid.toolchain import (CythonRecipe, shprint,
+                                        current_directory, info, Recipe)
+from pythonforandroid.patching import will_build, check_any
+import sh
+from os.path import join
+
+
+class Libxml2Recipe(CythonRecipe):
+    version = '2.9.4'
+    url = 'ftp://xmlsoft.org/libxml2/libxml2-{version}.tar.gz'
+    name = 'libxml2'
+    depends = [('python2', 'python3crystax')]
+
+    def get_recipe_env(self, arch):
+        env = super(Libxml2Recipe, self).get_recipe_env(arch)
+
+        try:
+            sh.sed('runtest\$(EXEEXT) \\/ \\/', 'Makefile')
+        except:
```

Comment:
What are these try/except blocks for? If you expect a specific error, catch it explicitly, catching everything is bad style since you'll find it hard to debug problems that you didn't expect as they're swallowed the same way as the one you did anticipate. Since the old toolchain didn't expect any errors, you probably just don't want the try/except at all - if it fails, it's not for an expected reason so you can let p4a crash.